code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -559,6 +559,16 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 				qdel(I)
 		return
 
+/datum/materialProc/ice_temp
+	execute(var/atom/owner, var/temp)
+		if(!prob(temp - T0C)) return //percentage chance to melt for every degree above freezing
+		if(ON_COOLDOWN(owner, "ice_temp", 10 SECONDS))
+			return
+		var/turf/tloc = get_turf(owner)
+		tloc.visible_message(SPAN_ALERT("[owner] melts due to heat!"))
+		qdel(owner)
```

Comment:
Maybe spawn a small amount of water here? Water is relatively harmless and easy to flood with anyway so it should be fine?

Reply:
Since mining can end up making a lot of ice chunks when mining certain asteroids and since explosions (also commonly found in mining) expose exploded items to heat, this has the potential to spawn quite a lot of water pools. My understanding is that lots of chempools is rather unperformant.